apps/syntrest/src/main/java/no/nav/registre/syntrest/kubernetes/KubernetesController.java

```diff
@@ -1,167 +1,125 @@
                                 @Value("${isAlive}") String isAliveUrl,
                                 @Value("${docker-image-path}") String dockerImagePath,
                                 @Value("${max-alive-retries}") int maxRetries,
-                                @Value("${alive-retry-delay}") int retryDelay) {
+                                @Value("${alive-retry-delay}") int retryDelay
+            ) {
 
-        this.manifestPath = "/nais/{appName}.yaml";
-        this.isAliveUri = new UriTemplate(isAliveUrl);
-        this.dockerImagePath = dockerImagePath;
+        this.naisYaml = naisYaml;
+        this.gitHubConsumer = gitHubConsumer;
+        this.isAliveUrl = isAliveUrl;
         this.maxRetries = maxRetries;
         this.retryDelay = retryDelay;
         this.api = customObjectsApi;
 
-
-        // nullcheck proxy port/url -> settes i run configuration på utvimage.
-
-
-        if (!"local".equals(proxyUrl) && proxyPort != 0) {
-            this.authRestTemplate = restTemplateBuilder
-                .requestFactory(() -> new HttpComponentsClientHttpRequestFactory(HttpClientBuilder.create()
-                        .setRoutePlanner(new DefaultProxyRoutePlanner(new HttpHost(proxyUrl, proxyPort)))
-                        .build()))
-                .additionalInterceptors(new BasicAuthenticationInterceptor(github_username, github_password))
-                .build();
-        } else {
-            this.authRestTemplate = restTemplateBuilder
-                .requestFactory(() -> new HttpComponentsClientHttpRequestFactory(HttpClientBuilder.create().build()))
-                .additionalInterceptors(new BasicAuthenticationInterceptor(github_username, github_password))
-                .build();
-        }
-
-        this.noAuthRestTemplate = restTemplateBuilder.requestFactory(() -> new HttpComponentsClientHttpRequestFactory(HttpClientBuilder.create()
-                .setRoutePlanner(new SystemDefaultRoutePlanner(ProxySelector.getDefault()))
-                .setSSLHostnameVerifier(new DefaultHostnameVerifier())
-                .setDefaultRequestConfig(RequestConfig.custom()
-                        .setConnectTimeout(TIMEOUT)
-                        .setSocketTimeout(TIMEOUT)
-                        .setConnectionRequestTimeout(TIMEOUT)
-                        .build())
-                .setMaxConnPerRoute(2000)
-                .setMaxConnTotal(5000)
-                .build())).build();
+        HttpClient httpClient = HttpClient.create()
+                .responseTimeout(Duration.ofMillis(TIMEOUT));
+        ReactorClientHttpConnector connector = new ReactorClientHttpConnector(httpClient);
+        this.webClient = WebClient.builder().clientConnector(connector).build();
     }
 
     public void deployImage(String appName) throws ApiException, InterruptedException {
-        Map<String, Object> manifestFile = prepareYaml(appName);
+        String imageTag = gitHubConsumer.getApplicationTag(appName);
```

Comment:
Hva hvis `getApplicationTag` feiler og `imageTag` er lik ""? Burde kanskje ha en sjekk.

Reply:
eller bare kaste en feil? Regner med at det ikke går ann å feilhåntere?

Reply:
Refaktorerer til at GitHubConsumer kaster
`throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Could not find application tag.");`
dersom tag ikke blir funnet.